### PR TITLE
Reduce the code required for integration testing

### PR DESCRIPTION
**Pull request by LukeButters**
# Background

Makes the integration tests smaller making it easier to focus on what is being tested.

# Results

slightly smaller tests.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests.Integration/PollingTentacleTest.cs [hunk 1]
```diff
@@ -64,32 +60,28 @@ public async Task BasicCommunicationsWithWithAnOldPollingTentacle()
...
 
                     var res = tentacleClient.ScriptService.GetStatus(new ScriptStatusRequest(new ScriptTicket("1212"), 111));
                     res.ExitCode.Should().Be(0);
                 }, cts.Token);
 
-                await Task.WhenAny(runningTentacle, testTask);
+                await Task.WhenAny(runningTentacle.Task, testTask);
                 cts.Cancel();
-                await Task.WhenAll(runningTentacle, testTask);
+                await Task.WhenAll(runningTentacle.Task, testTask);
```
Comment (LukeButters) [inline source/Octopus.Tentacle.Tests.Integration/PollingTentacleTest.cs:84]: I think I might do what is suggested [here](https://github.com/OctopusDeploy/OctopusTentacle/pull/470/files#r1185710292) to remove the need to do this little dance.

File: source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs [hunk 2]
```diff
@@ -35,38 +37,83 @@ public PollingTentacleBuilder WithTentacleExe(string tentacleExe)
...
+            
+            var runningTentacle =  Task.Run(() =>
             {
                 try
                 {
-                    RunTentacleCommand(tentacleExe, new[] {"agent", "--config", configFilePath, $"--instance={instanceName}", "--noninteractive"}, tmp, cancellationToken);
+                    RunTentacleCommandOutOfProcess(tentacleExe, new[] {"agent", "--config", configFilePath, $"--instance={instanceName}", "--noninteractive"}, tmp, 
+                        s =>
+                        {
+                            if (s.Contains("Agent will not listen") || s.Contains("Agent listening on"))
```
Comment (LukeButters) [inline source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs:92]: In theory this should support both listening and polling.

File: source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs [hunk 3]
```diff
@@ -3,6 +3,8 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Autofac;
+using Nito.AsyncEx;
+using Nito.AsyncEx.Interop;
```
Comment (acodrington) [inline source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs:7]: These look like an accidental addition, though I may be be mistaken.

File: source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs [hunk 4]
```diff
@@ -29,44 +34,99 @@ public PollingTentacleBuilder WithTentaclePollSubscription(Uri tentaclePollSubsc
...
-            AddCertificateToTentacle(tentacleExe, configFilePath, instanceName, Certificates.TentaclePfxPath, tempDirectory, cancellationToken);
+            AddCertificateToTentacle(tentacleExe, configFilePath, instanceName, CertificatePfxPath, tempDirectory, cancellationToken);
             ConfigureTentacleToPollOctopusServer(
                 configFilePath,
                 octopusHalibutPort,
                 octopusThumbprint ?? Certificates.ServerPublicThumbprint,
-                tentaclePollSubscriptionId ?? PollingSubscriptionId.Generate());
+                subscriptionId);
+
+            CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```
Comment (acodrington) [inline source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs:67]: Why do we need to create a new source if we've already been passed a token? Is it about stack traces upon cancellation or something?

**Review by acodrington — APPROVED** [review]
Love it, much cleaner, lets us focus on testing the functionality rather than the setup 👍🏻

File: source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs (hunk 3 above)
Comment (LukeButters) [reply source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs:6]: The second is needed but not the first. Will clean up

**Pull request merged**